src/components/VideoDetailsRegularView/VideoDetailsRegularView.jsx

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useContext, useEffect, useState } from 'react';
+import useRequest from '../../clients/useRequest';
+import {
+  youtubeApiGetSpecificVideo,
+  youtubeApiSearchRelated,
+} from '../../clients/youtube-api';
+import { GlobalContext } from '../../contexts/GlobalContextProvider';
+import { notEmpty } from '../../utils/typeHelper';
+import {
+  standadizeRelatedVideos,
+  standadizeSingleVideo,
+} from '../../utils/youtubeDataHelper';
+import AnnounceText from '../AnnounceText';
+import VideoDetailsView from '../VideoDetailsView';
+
+export default function VideoDetailsRegularView(props) {
+  const { videoId } = props;
+
+  const [videoInfo, setVideoInfo] = useState(null);
+  const [relatedVideos, setRelatedVideos] = useState([]);
+
+  const [globalState, globalDispatch] = useContext(GlobalContext);
+
+  const isLoggedIn = notEmpty(globalState.user);
+
+  const getSpecificVideoRequest = useRequest();
+  const searchRelatedRequest = useRequest();
+
+  useEffect(() => {
+    getSpecificVideoRequest.handleFetchPromise(youtubeApiGetSpecificVideo(videoId));
+    searchRelatedRequest.handleFetchPromise(youtubeApiSearchRelated(videoId));
+  }, []);
+
+  useEffect(() => {
+    getSpecificVideoRequest.handleFetchPromise(youtubeApiGetSpecificVideo(videoId));
+    searchRelatedRequest.handleFetchPromise(youtubeApiSearchRelated(videoId));
+  }, [videoId]);
```

Comment:
Only this is enough, the current code duplicates API calling.

```js
useEffect(() => {
    getSpecificVideoRequest.handleFetchPromise(youtubeApiGetSpecificVideo(videoId));
    searchRelatedRequest.handleFetchPromise(youtubeApiSearchRelated(videoId));
  }, [videoId]);
```

The difference between `[]` and `[videoId]` is:
- `[]` runs only once when the component is rendered
- `[videoId]` runs the first time when component is rendered and whenever `videoId` changes